File: 02-programacion-basica/01-entorno-y-entrada-salida/scala/soluciones/RetoFichaEstudiante.scala
import scala.io.StdIn.readLine

@main def retoFichaEstudiante(): Unit =
  // readLine conecta la solución con la entrada estándar.
  val nombre = readLine("Nombre: ").trim
  val edad = readLine("Edad: ").trim.toInt
  val carrera = readLine("Carrera: ").trim
  println(s"\nNombre: $nombre\nEdad: $edad\nCarrera: $carrera")

File: 02-programacion-basica/01-entorno-y-entrada-salida/java/soluciones/RetoFichaEstudiante.java
import java.util.Scanner;

public class RetoFichaEstudiante {
    public static void main(String[] args) {
        // Scanner conecta el programa con la entrada estándar (teclado).
        try (Scanner scanner = new Scanner(System.in)) {
            System.out.print("Nombre: ");
            String nombre = scanner.nextLine().trim();
            System.out.print("Edad: ");
            int edad = Integer.parseInt(scanner.nextLine().trim());
            System.out.print("Carrera: ");
            String carrera = scanner.nextLine().trim();

            System.out.printf("%nNombre: %s%nEdad: %d%nCarrera: %s%n", nombre, edad, carrera);
        }
    }
}
